zio-cache/shared/src/main/scala/zio/cache/Lookup.scala

```diff
@@ -12,12 +12,49 @@ import zio.ZIO
  * given a key. Given any effectual function you can convert it to a lookup
  * function for a cache by using the `Lookup` constructor.
  */
-final case class Lookup[-Key, -Environment, +Error, +Value](lookup: Key => ZIO[Environment, Error, Value])
-    extends (Key => ZIO[Environment, Error, Value]) {
+final class Lookup[-Key, -Environment, +Error, +Value] private (
+  private val lookup: Iterable[Key] => ZIO[Environment, Error, Chunk[Value]]
+) extends (Key => ZIO[Environment, Error, Value]) {
 
   /**
    * Computes a value for the specified key or fails with an error.
    */
   def apply(key: Key): ZIO[Environment, Error, Value] =
-    lookup(key)
+    lookup(Chunk(key)).map(_.head)
```

Comment:
Similar to my previous comment, the `lookup` function is user-supplied and it might even be considered legit that it'd return an empty collection when the `key` doesn't exist. So we should use `headOption` here to ensure safety. This should not be an issue because `Cache` impl doesn't use this overload, assuming existing users don't use `Lookup` directly. If we make the change and break compatibility, it's still a correct thing to do. :)